src/main/java/com/nike/cerberus/endpoints/secret/SecureDataEndpointV1.java

```diff
@@ -0,0 +1,204 @@
+        if (! securityContext.isPresent() || ! (securityContext.get().getUserPrincipal() instanceof CerberusPrincipal)) {
+            log.error("Security context was null or principal was not instance of Cerberus Principal");
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        SecureDataRequestInfo requestInfo = parseInfoFromPath(request.getPath());
+
+        // if that path was invalid throw Vault style permission denied error
+        if (! doesRequestHaveRequiredParams(requestInfo.getCategory(), requestInfo.getSdbSlug())) {
+            log.error("Required path params missing, PATH: {}", request.getPath());
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        CerberusPrincipal principal = (CerberusPrincipal) securityContext.get().getUserPrincipal();
+
+        String sdbBasePath = String.format("%s/%s/", requestInfo.getCategory(), requestInfo.getSdbSlug());
+        Optional<String> sdbId = safeDepositBoxService.getSafeDepositBoxIdByPath(sdbBasePath);
+
+        // check that the principal has the proper permissions or throw Vault style permission denied error
+        if (! sdbId.isPresent() || ! permissionService.doesPrincipalHavePermission(principal, sdbId.get(),
+                        SecureDataAction.fromMethod(request.getMethod()))) {
+            log.info("SDB Id not found or permission was not granted for principal, principal: {}, sdb: {}," +
+                    " path: {}", principal.getName(), sdbId.orElseGet(() -> "missing"), request.getPath());
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        requestInfo.setSdbid(sdbId.get());
+
+        return doExecute(requestInfo, request, longRunningTaskExecutor, ctx, securityContext.get());
+    }
+
+    private boolean doesRequestHaveRequiredParams(String category, String sdbSlug) {
+        return isNotBlank(category) || isNotBlank(sdbSlug);
+    }
+
+    protected SecureDataRequestInfo parseInfoFromPath(String path) {
+        String[] parts = path.split("/", 6);
```

Comment:
Probably should have named constants for all of these values.  And a unit test just to make sure nobody ever breaks.